pvlib/test/test_way_faster.py

```diff
@@ -0,0 +1,82 @@
+"""
+testing way faster single-diode methods using JW Bishop 1988
+"""
+
+from time import clock
+import logging
+import numpy as np
+from pvlib import pvsystem
+from pvlib.way_faster import faster_way
+
+logging.basicConfig()
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel(logging.DEBUG)
+
+POA = 888
+TCELL = 55
+CECMOD = pvsystem.retrieve_sam('cecmod')
+
+
+def test_spr_e20_327():
+    spr_e20_327 = CECMOD.SunPower_SPR_E20_327
+    x = pvsystem.calcparams_desoto(
+        poa_global=POA, temp_cell=TCELL,
+        alpha_isc=spr_e20_327.alpha_sc, module_parameters=spr_e20_327,
+        EgRef=1.121, dEgdT=-0.0002677)
+    il, io, rs, rsh, nnsvt = x
+    tstart = clock()
+    pvs = pvsystem.singlediode(*x)
+    tstop = clock()
+    dt_slow = tstop - tstart
+    LOGGER.debug('single diode elapsed time = %g[s]', dt_slow)
+    tstart = clock()
+    out = faster_way(*x, log=False, test=False)
+    isc, voc, imp, vmp, pmp, _, _ = out.values()
+    tstop = clock()
+    dt_fast = tstop - tstart
+    LOGGER.debug('way faster elapsed time = %g[s]', dt_fast)
+    LOGGER.debug('spr_e20_327 speedup = %g', dt_slow / dt_fast)
+    assert np.isclose(pvs['i_sc'], isc)
+    assert np.isclose(pvs['v_oc'], voc)
+    # the singlediode method doesn't actually get the MPP correct
+    pvs_imp = pvsystem.i_from_v(rsh, rs, nnsvt, vmp, io, il)
+    pvs_vmp = pvsystem.v_from_i(rsh, rs, nnsvt, imp, io, il)
+    assert np.isclose(pvs_imp, imp)
+    assert np.isclose(pvs_vmp, vmp)
+    assert np.isclose(pvs['p_mp'], pmp)
+    return isc, voc, imp, vmp, pmp, pvs
+
+
+def test_fs_495():
+    fs_495 = CECMOD.First_Solar_FS_495
+    x = pvsystem.calcparams_desoto(
+        poa_global=POA, temp_cell=TCELL,
+        alpha_isc=fs_495.alpha_sc, module_parameters=fs_495,
+        EgRef=1.475, dEgdT=-0.0003)
+    il, io, rs, rsh, nnsvt = x
+    x += (101, )
+    tstart = clock()
+    pvs = pvsystem.singlediode(*x)
+    tstop = clock()
+    dt_slow = tstop - tstart
+    LOGGER.debug('single diode elapsed time = %g[s]', dt_slow)
+    tstart = clock()
+    out = faster_way(*x, log=False, test=False)
+    isc, voc, imp, vmp, pmp, _, _, i, v, p = out.values()
+    tstop = clock()
+    dt_fast = tstop - tstart
+    LOGGER.debug('way faster elapsed time = %g[s]', dt_fast)
+    LOGGER.debug('fs_495 speedup = %g', dt_slow / dt_fast)
+    assert np.isclose(pvs['i_sc'], isc)
+    assert np.isclose(pvs['v_oc'], voc)
+    # the singlediode method doesn't actually get the MPP correct
```

Comment:
Is this kind of a bigish deal?

Reply:
I calculate a ~0.02% (absolute) difference between my I_mp_A and V_mp_V values (pvfit) and the pvlib ones for the SunPower module.

pvlib calculation:
OrderedDict([('i_sc', 5.857594583158265), ('v_oc', 57.779106188588685), ('i_mp', 5.374898118822274), ('v_mp', 47.473396494199655), ('p_mp', 255.16466951077766), ('i_x', 5.770970666689348), ('i_xx', 4.048918484984828)])

pvfit calculation:
ff=0.7539291803981164, i_sc_A=5.857594583158266, i_mp_A=5.375969556815134, p_mp_W=255.16475223820953, v_mp_V=47.46395037054039, v_oc_V=57.77910618858893
1.9202373104812607

Reply:
Doesn't this change in this PR, because default method is no longer LambertW?